infra/src/main/kotlin/fi/fta/geoviite/infra/publication/Publication.kt

```diff
@@ -674,3 +674,5 @@ data class PreparedPublicationRequest(
     val message: FreeTextWithNewLines,
     val cause: PublicationCause,
 )
+
+data class RatkoPlanItemId(val id: Int)
```

Comment:
Koska tämä on tällä puolella, tässä lienee kyse meidän sisäisen logiikan tyypistä? Tämä voisi varmaan olla vain IntId, joskin ei se sinällään haittaa eriyttää näitä spesifimmäksikään näin.

Reply:
Taitaa olla parempi ratkaisu, että siirrän tämän RatkoCommonModeliin, koska kyllä tämä aika puhtaasti tarkoittaa Ratkon kanssa toimivaa käytöstä. Tässä pullarissa tämän luokan merkitys on kylläkin vielä vähän hatara, mutta olen lisäämässä tähän JsonCreator- ja JsonValue-annotaatiot niin, että RatkoPlanItem-luokka sisältää RatkoPlanItemId:n, joka sitten serialisoituu pelkäksi intiksi; ja sittenhän sen käytös eriääkin IntId:n käytöksestä niin, että kyllähän sen on oltava oma luokkansa.

Tai ehkä jopa voisi tehdä jonkin PureIntId tjsp. -luokan, joka olisi geneerinen luokka käsittelemään kaikkia tapauksia, missä konseptuaalisesti puhutaan jonkin olion tunnisteesta, ja tämä tunniste serialisoituu JSONissa puhtaaksi kokonaisluvuksi.